### PR TITLE
Remove invalid async-flag from syntax error tests

### PR DESCRIPTION
**Pull request by anba**

### CODE REVIEW

File: src/params/syntax/async-gen-named-func-expr.template [hunk 1]
```diff
@@ -13,7 +13,6 @@ info: |
         7. Let closure be ! AsyncGeneratorFunctionCreate(Normal, FormalParameters,
            AsyncGeneratorBody, funcEnv, strict).
         [...]
-flags: [async]
```
Comment (bakkot) [inline src/params/syntax/async-gen-named-func-expr.template]: This one doesn't look like a syntax error test to me.

Comment (anba) [inline src/params/syntax/async-gen-named-func-expr.template]: It's currently only used for syntax error tests. The real issue is that the template generates tests with the "async" flag, but doesn't call $DONE.

Comment (leobalter) [inline src/params/syntax/async-gen-named-func-expr.template]: all the respective cases using templates from the `syntax` folder have the negative flag, i.e.:

```yaml
negative:
  phase: early
  type: SyntaxError
```

**Comment by leobalter** [comment]
@anba if you don't mind, we try to split commits from the actual changes to the generated codes. This helps identifying the manual from generated changes.

I'm merging this as is. The changes here are small and not worth holding this PR open.

Thank you so much for finding these nit picks.

**Comment by anba** [comment]
Okay, I'll try to remember splitting the commits next time.

**Pull request merged**